### ISSUE TITLE
Disallow submissions with a guid > 255 characters

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#11631]
### Describe the problem and steps to reproduce it:
(Please include as many details as possible.)
- Submit an add-on that has an `id` in its manifest that obey the `@foo` format but is longer than 255 characters 

### What happened?
- Validation passes, but you get a 500 error (https://sentry.prod.mozaws.net/operations/olympia-dev/issues/5942419/?query=is:unresolved)

### What did you expect to happen?
- You should get a validation error, because even though we now accept long ids, the column in the database has a 255 character limit.

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)
- The limit should be applied even to unlisted add-ons. Maybe it would be simpler to just do this in the linter ?

**Comment by diox** [comment]
cc @EnTeQuAk after filing that I realized that maybe we wanted to have that in the linter instead. What do you think ?

**Comment by EnTeQuAk** [comment]
Agreed this should be in the linter. We shouldn't raise a server error though, so if there's an easy way to avoid that and throw a proper error in the linter that'd be nice I think
